### PR TITLE
Add built-in support for basic auth

### PR DESCRIPTION
**Pull request by danra**
This allows using the encoded token for basic auth-based git access.

This is useful for scripts expecting to be able to access private git repos. My specific case was using [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake) in a repo that brings in other private repos as dependencies.

While the encoding to make this work is very simple, the documentation for it is sparse and it is not trivial to figure out, so there's value in providing this out of the box.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,11 +99,37 @@ jobs:
...
         with:
           token: ${{ steps.app-token.outputs.token }}
-          issue-number: ${{ github.event.issue.number }}
-          body: "Hello, World!"
+          ref: ${{ github.head_ref }}
+          # Access allowed to private submodules in the current owner's installation
+          submodules: recursive
+          # Make sure the value of GITHUB_TOKEN will not be persisted in repo's config
+          persist-credentials: false
+```
```
Comment (gr2m) [inline README.md:110]: Please remove unrelated changes

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -99,11 +99,37 @@ jobs:
           app-id: ${{ vars.APP_ID }}
           private-key: ${{ secrets.PRIVATE_KEY }}
           owner: ${{ github.repository_owner }}
-      - uses: peter-evans/create-or-update-comment@v3
+      - uses: actions/checkout@v4
```
Comment (gr2m) [inline README.md:102]: please revert

**Review by gr2m — CHANGES_REQUESTED** [review]
I would not add this feature to this action, but we can document the workflow in the README.

It's similar to this pull request: #137 which we closed in favor of #142

I would suggest to add a step which pipes the string `x-access-token:${{ steps.app-token.outputs.token }} | bash64` and writes that to a new output as in did in #142

In future, can you please create an issue first before a pull request? We appreciate the effort you put into it, but having a discussion first might help prevent unnecessary work and frustration.

File: README.md (hunk 2 above)
Comment (parkerbxyz) [reply README.md:102]:
```suggestion
      - uses: peter-evans/create-or-update-comment@v4
```

**Comment by danra** [comment]
> I would suggest to add a step which pipes the string x-access-token:${{ steps.app-token.outputs.token }} | bash64 

Unfortunately, that's not cross-platform (doesn't work on Windows).
I used [github-script](https://github.com/actions/github-script) for cross-platform base64 encoding:

```yaml
- name: Prepare basic auth credentials
  id: basic-auth-creds
  uses: actions/github-script@v7
  with:
    script: return btoa("x-access-token:${{ steps.app-token.outputs.token }}")
    result-encoding: string

- name: Checkout with private submodules
  env:
    GIT_CONFIG_COUNT: 1
    GIT_CONFIG_KEY_0: http.https://github.com/.extraheader
    GIT_CONFIG_VALUE_0: "AUTHORIZATION: basic ${{ steps.basic-auth-creds.outputs.result }}"
  # Access allowed to private submodules in the current owner's installation
  run: git clone --recurse-submodules --shallow-submodules https://github.com/${{ github.repository }}
```

> In future, can you please create an issue first before a pull request? We appreciate the effort you put into it, but having a discussion first might help prevent unnecessary work and frustration.

Sure, no worries. I'll close the PR. A README addition would be helpful, indeed.
